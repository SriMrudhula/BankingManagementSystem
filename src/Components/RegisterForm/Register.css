.err1{
    font-size: 15px;
    color: red;
}

.img1{
    background-image: url('../../Asserts/Images/banking.jpg');
    height: 865px;
    background-repeat: no-repeat;
    background-size: cover;
    margin-top: -90px;
    width:100%;
    background-attachment: fixed;
}
.btn-primary{
    width:100px;
    height: 50px;
    border-radius: 15px;
}
.b1{
    display:inline-block;
    width: 600px;
    height: 750px;
    padding: 20px;
    background-color: rgba(0,0,0,0.5);
    background-size: 400px 480px;   
    margin-top:100px;
    position: relative;
     } 
     label{
       font-size: large;
        color: white;
    }
    input{
        height:50px;
        width:100px;
    }
    select{
        font: inherit;
    }
    .heading{
        color: white;
        font-size: 30px;
    }
