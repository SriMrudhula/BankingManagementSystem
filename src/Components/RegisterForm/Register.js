import React, { Component } from 'react';
import './Register.css';
import axios from 'axios';
import { Col, Row, Button } from 'reactstrap';
import { Container } from '@material-ui/core';
const statesByCountry = {
    India: ["Andhra Pradesh",'TamilNadu',"Chennai","Kerala","Maharashtra","Karnataka"],
    USA: ["Texas","Washington","Virginia","California","Florida"],
    Australia:["Western Australia","south Australia","Tasmania","Queensland","New South Wales"],
    Canada:["Alberta","British Columbia","Manitoba","New Brunswick","Newfoundland and Labrador","Northwest Territories","Nova Scotia"],
    Denmark:["North Zealand","East Zealand"]
    }
class Register extends Component
{
    state={    
        name:"",
        username:"",
        password:"",
        confirmPwd:"",
        address:"",
        country:"",
        state:"",
        email:"",
        contact:"",
        dob:"",
        accType:"",
        branchName:"",
        initialDeposit:"",
        identificationProofType:"",
        nameError:"",
        emailError:"",
        contactError:"",
        usernameError:"",  
        AccountType:false,
        duplicateUser:false,  
            users:[]
    }
    constructor() 
    {
        super();
        localStorage.removeItem("id");
    }

    componentDidMount(){
        axios.get("http://localhost:3000/users")
        .then(response =>{
            this.state.users=response.data
        })
        .catch(error=>{
            console.log(error);
        })
    }

    handleChange =event=>
    {
        this.setState({
            [event.target.name]:event.target.value
        });
    };

    duplicateUsername=()=>{
        for(let i=0;i<this.state.users.length;i++)
        {
            if(this.state.users[i].username===this.state.username)
             {
                this.state.duplicateUser=true;
                 break;
             }
             else
                this.state.duplicateUser=false;          
             }      
            if(this.state.duplicateUser===true)
                return true;
            else 
                return false;
        }

    validate = () => {
        let nameError=""
        let emailError=""
        let contactError=""
        let usernameError=""
        const isDuplicate=this.duplicateUsername();
        if(isDuplicate)
           usernameError="Username already Exists"
        if(!(this.state.email.includes("@") && this.state.email.includes(".com")))
           emailError='Please enter correct email';
        // if(!this.state.name.includes("^[a-zA-Z]+$"))
        //    nameError="Please enter Username"
        if(this.state.contact.length!==10)
            contactError="Please Enter Correct Contact"
        if(emailError!=="" || nameError!=="" || contactError!=="" || usernameError!=="" ){
            this.setState({nameError,emailError,contactError,usernameError});
            return false;
            }
            return true;
    }

    makeSubmenu = event => 
    {
        this.setState({[event.target.name]:event.target.value});
        event=document.getElementById("countrySelect").value;
        if(event.length===0)
            document.getElementById("stateSelect").innerHTML = "<option></option>";
        else 
        {
            var stateOptions = "";
            stateOptions+="<option>Choose State</option>"
            for(let stateId in statesByCountry[event])
            {
                stateOptions+="<option>"+statesByCountry[event][stateId]+"</option>";
            }
            document.getElementById("stateSelect").innerHTML = stateOptions;
        }
    }

    setAccountType =event => 
    {
        this.setState({[event.target.name]:event.target.value});
        event=new Date(document.getElementById('dob').value);
        var year=event.getFullYear();
        var age=2020-parseInt(year,10);
        var account=""
        account="<option>Account Type</option><option>Savings Account</option>"
        if(age>18)
            account+="<option>Salary Account</option>";
        document.getElementById("accType").innerHTML = account;
    }

    handleSubmit = event => 
    {
        event.preventDefault();
        const isValid=this.validate();
        if(isValid)
        {
            if(this.state.password!==this.state.confirmPwd)
                alert("Password and Confirm Password are not same");
            else
            {
                let user={name:this.state.name,username:this.state.username,password:this.state.password,confirmPwd:this.state.confirmPwd,
                address:this.state.address,country:this.state.country,state:this.state.state,email:this.state.email,contact:this.state.contact,
                dob:this.state.dob,accType:this.state.accType,branchName:this.state.branchName,initialDeposit:this.state.initialDeposit,
                identificationProofType:this.state.identificationProofType,id:Math.floor(Math.random()*100)+1}
            axios.post('http://localhost:3000/users',user)
            .then(respone =>{
            this.handleClick("login");  
            })
            .catch(error=>{
            console.log(error)
            })
            }
        }
    }
    _onFocus=(event)=>{
        event.currentTarget.type = "date";
    }

    handleLogin=()=>
    {
        this.handleClick("login");  
    }

    handleClick = event => 
    {
        this.props.handleClick(event);     
    }

    render()
    {
        return (
        <div className="img1">
        <div className="b1">
        <div className="heading">Register</div><br/>
        <Container>
        <form onSubmit={this.handleSubmit} >
            <div className="form-group">
                <Row>
                    <Col xs={6} >
        <input name="name" className="form-control" placeholder="Enter Name" type="text" value={this.state.name} onChange={this.handleChange} required/>
        <div className="err1">{this.state.nameError}</div>
        </Col>
        <Col xs={6}>
        <input name="username" type="text" className="form-control" placeholder="Enter Username" value={this.state.username} onChange={this.handleChange} required/>
        <div className="err1">{this.state.usernameError}</div>
        </Col>
        </Row><br/>
        <Row>
        <Col>
        <input type="password" name="password" placeholder="Enter Password" className="form-control" value={this.state.password} onChange={this.handleChange} required/>
        </Col>
        <Col>
        <input type="password" name="confirmPwd" placeholder="Enter Confirm Password" value={this.state.confirmPwd} className="form-control" onChange={this.handleChange} required/>
        </Col>
        </Row><br/>
        <Row>
        <Col>
        <select id="countrySelect" onChange={this.makeSubmenu} className="form-control" name="country" value={this.state.country} required>
        <option value="">Choose Country</option>
        <option value="India">India</option>
        <option value="USA">USA</option>
        <option value="Australia">Australia</option>
        <option value="UK">UK</option>
        <option value="Canada">Canada</option>
        <option value="Denmark">Denmark</option>
        </select>
        </Col>
        <Col>
        <select id="stateSelect" size="1" name="state" className="form-control" value={this.state.state} onChange={this.handleChange.bind(this)} required >
        <option>Choose State</option>
        </select>
        </Col>
        </Row><br/>
        <Row>
        <Col xs={6}>
        <input text="email" value={this.state.email} placeholder="Enter Email Id" name="email" className="form-control" onChange={this.handleChange} required/>
        <div className="err1">{this.state.emailError}</div><br/>
        </Col>
        <Col xs={6}>
        <input type="number" value={this.state.contact} placeholder="Enter Contact Number" name="contact" className="form-control" onChange={this.handleChange} required/>
        <div className="err1">{this.state.contactError}</div><br/>
        </Col></Row>
        <Row>
        <Col>
        <input type="text" id="dob" placeholder="Date of Birth" onFocus={this._onFocus} value={this.state.dob} className="form-control" max={new Date().toISOString().split("T")[0]} onChange={this.setAccountType} name="dob" required/>
        </Col>
        <Col>
        <select  value={this.state.accType} id="accType" className="form-control" name="accType" onChange={this.handleChange} required>
        <option>Account Type</option>
        </select>
        </Col>
        </Row><br/>
        <Row>
        <Col>
        <input type="text" value={this.state.branchName} placeholder="Enter Branch Name" className="form-control" name="branchName" onChange={this.handleChange} required/>
        </Col>
        <Col>
        <input type="number" value={this.state.initialDeposit} placeholder="Enter Initial Deposit" className="form-control" name="initialDeposit" onChange={this.handleChange} required/>
        </Col>
        </Row><br/>
        <Row>
        <textarea value={this.state.address} placeholder="Enter Address" name="address" className="form-control" onChange={this.handleChange}></textarea>
        </Row><br/>
        <Row>
        <input type="text" value={this.state.identificationProofType} placeholder="Enter Identification Proof type" className="form-control" name="identificationProofType" onChange={this.handleChange} required/> 
        </Row><br/>
        <Button className="Btn" id="registersubmit" color="primary">Register</Button>
        </div>
        </form>
        Already Registered <button onClick={this.handleLogin}>Login</button> 
        </Container>
        </div>
        </div>
        )
    }
}
export default Register;