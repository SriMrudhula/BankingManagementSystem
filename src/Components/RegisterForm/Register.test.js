import * as React from 'react';
import Enzyme,{shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom';
import Register from './Register';
Enzyme.configure({ adapter: new Adapter() });
describe('Test when form is submitted', () => {
    it('should submit form data', () => {
        let form = shallow(<Register/>); 
        form.instance().handleSubmit = jest.fn();
        form.instance().handleSubmit();
        form.update(); 
        form.find('#registersubmit').simulate('submit');


        expect(form.find('#registersubmit').length).toEqual(1);
        expect(form.instance().handleSubmit).toHaveBeenCalled();
    });
    
    it('renders correctly', () => {
        const tree = renderer.create(<BrowserRouter><Register/></BrowserRouter>).toJSON();
        expect(tree).toMatchSnapshot();
      });
});
