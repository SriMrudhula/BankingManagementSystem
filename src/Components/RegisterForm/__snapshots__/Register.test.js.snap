// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test when form is submitted renders correctly 1`] = `
<div
  className="img1"
>
  <div
    className="b1"
  >
    <div
      className="heading"
    >
      Register
    </div>
    <br />
    <div
      className="MuiContainer-root MuiContainer-maxWidthLg"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="form-group"
        >
          <div
            className="row"
          >
            <div
              className="col-6"
            >
              <input
                className="form-control"
                name="name"
                onChange={[Function]}
                placeholder="Enter Name"
                required={true}
                type="text"
                value=""
              />
              <div
                className="err1"
              >
                
              </div>
            </div>
            <div
              className="col-6"
            >
              <input
                className="form-control"
                name="username"
                onChange={[Function]}
                placeholder="Enter Username"
                required={true}
                type="text"
                value=""
              />
              <div
                className="err1"
              >
                
              </div>
            </div>
          </div>
          <br />
          <div
            className="row"
          >
            <div
              className="col"
            >
              <input
                className="form-control"
                name="password"
                onChange={[Function]}
                placeholder="Enter Password"
                required={true}
                type="password"
                value=""
              />
            </div>
            <div
              className="col"
            >
              <input
                className="form-control"
                name="confirmPwd"
                onChange={[Function]}
                placeholder="Enter Confirm Password"
                required={true}
                type="password"
                value=""
              />
            </div>
          </div>
          <br />
          <div
            className="row"
          >
            <div
              className="col"
            >
              <select
                className="form-control"
                id="countrySelect"
                name="country"
                onChange={[Function]}
                required={true}
                value=""
              >
                <option
                  value=""
                >
                  Choose Country
                </option>
                <option
                  value="India"
                >
                  India
                </option>
                <option
                  value="USA"
                >
                  USA
                </option>
                <option
                  value="Australia"
                >
                  Australia
                </option>
                <option
                  value="UK"
                >
                  UK
                </option>
                <option
                  value="Canada"
                >
                  Canada
                </option>
                <option
                  value="Denmark"
                >
                  Denmark
                </option>
              </select>
            </div>
            <div
              className="col"
            >
              <select
                className="form-control"
                id="stateSelect"
                name="state"
                onChange={[Function]}
                required={true}
                size="1"
                value=""
              >
                <option>
                  Choose State
                </option>
              </select>
            </div>
          </div>
          <br />
          <div
            className="row"
          >
            <div
              className="col-6"
            >
              <input
                className="form-control"
                name="email"
                onChange={[Function]}
                placeholder="Enter Email Id"
                required={true}
                text="email"
                value=""
              />
              <div
                className="err1"
              >
                
              </div>
              <br />
            </div>
            <div
              className="col-6"
            >
              <input
                className="form-control"
                name="contact"
                onChange={[Function]}
                placeholder="Enter Contact Number"
                required={true}
                type="number"
                value=""
              />
              <div
                className="err1"
              >
                
              </div>
              <br />
            </div>
          </div>
          <div
            className="row"
          >
            <div
              className="col"
            >
              <input
                className="form-control"
                id="dob"
                max="2020-10-09"
                name="dob"
                onChange={[Function]}
                onFocus={[Function]}
                placeholder="Date of Birth"
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="col"
            >
              <select
                className="form-control"
                id="accType"
                name="accType"
                onChange={[Function]}
                required={true}
                value=""
              >
                <option>
                  Account Type
                </option>
              </select>
            </div>
          </div>
          <br />
          <div
            className="row"
          >
            <div
              className="col"
            >
              <input
                className="form-control"
                name="branchName"
                onChange={[Function]}
                placeholder="Enter Branch Name"
                required={true}
                type="text"
                value=""
              />
            </div>
            <div
              className="col"
            >
              <input
                className="form-control"
                name="initialDeposit"
                onChange={[Function]}
                placeholder="Enter Initial Deposit"
                required={true}
                type="number"
                value=""
              />
            </div>
          </div>
          <br />
          <div
            className="row"
          >
            <textarea
              className="form-control"
              name="address"
              onChange={[Function]}
              placeholder="Enter Address"
              value=""
            />
          </div>
          <br />
          <div
            className="row"
          >
            <input
              className="form-control"
              name="identificationProofType"
              onChange={[Function]}
              placeholder="Enter Identification Proof type"
              required={true}
              type="text"
              value=""
            />
          </div>
          <br />
          <button
            aria-label={null}
            className="Btn btn btn-primary"
            id="registersubmit"
            onClick={[Function]}
          >
            Register
          </button>
        </div>
      </form>
      Already Registered 
      <button
        onClick={[Function]}
      >
        Login
      </button>
    </div>
  </div>
</div>
`;
